File: Player.java
public abstract class Player 
{ 
	private Field myField, enemyField;
	private String name;

	public Player(Field myField, Field enemyField, String name)
    {
        this.myField = myField;
        this.enemyField = enemyField;
        this.name = name;
    }
	public abstract void MakeMove();
	public abstract void ArrangeShips();   
    public Field getMyField()
    {
        return myField;
    }
    public void setMyField(Field myField)
    {
        this.myField = myField;
    }
    public Field getEnemyField()
    {
        return enemyField;
    }
    public void setEnemyField(Field enemyField)
    {
        this.enemyField = enemyField;
    }
    public String getName()
    {
        return name;
    }
    public void setName(String name)
    {
        this.name = name;
    }
}

File: Player.cpp
#include "Player.hpp"
Player::Player(Field& myField, Field& enemyField, std::string name) : m_myField(myField), m_enemyField(enemyField), m_name(name)
{
}
Field& Player::GetMyField()
{
	return m_myField;
}
void Player::SetMyField(Field& myField)
{
	m_myField = myField;
}
Field& Player::GetEnemyField()
{
	return m_enemyField;
}
void Player::SetEnemyField(Field& enemyField)
{
	m_enemyField = enemyField;
}
std::string Player::GetName()
{
	return m_name;
}
void Player::SetName(std::string name)
{
	m_name = name;
}
void Player::MakeMove()
{

}
void Player::ArrangeShips()
{

}